### PR TITLE
Speed up rendering the trip detail page

### PR DESCRIPTION
**Pull request by rwangsc18**
## Description
Clearly describe changes in your PR, new feature or bug fixes.
The function planner.getTripFromPlace() sends a http request to retrieve the photo URL for a destination. Multiple requests are sent for different places in a plan. This can lead to increased latency.

## Solution
* Describe your code changes in detail for reviewers as well as documentation purposes. 
Use Goroutines to speed up the processing to get plan details page. Tested locally and see the noticeable 60% of latency reduction, the average latency goes from 200ms to 80ms on my local laptop.

Previous latency to get the plan detail page, see screenshot
Screen Shot 2023-01-07 at 5 03 53 PM

Current latency to get the plan detail page, see screenshot
Screen Shot 2023-01-07 at 5 09 38 PM

## Testing
- [x] Integration testing on Heroku staging
- [ ] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: planner/planner.go [hunk 1]
```diff
@@ -465,13 +467,15 @@ func (planner *MyPlanner) getPlanDetails(c *gin.Context) {
 			return
 		}
 
-		placeDetails := planner.getTripFromPlace(cachePlaceDetails)
+		// planner.getTripFromPlace is a blocking call with http request
+		// send data via channel for async processing
+		channel <- planner.getTripFromPlace(cachePlaceDetails)
```
[the remark this answers is not in the document]
Comment (rwangsc18) [reply planner/planner.go]: You are right. Used `sync.WaitGroup()` plus go command to trigger Goroutines. Now I do see some noticeable latency drop (200ms -> 80ms) in getting trip detail page data.

File: planner/planner.go [hunk 2]
```diff
@@ -450,13 +451,15 @@ func (planner *MyPlanner) getPlanDetails(c *gin.Context) {
...
-		PlaceDetails:      make([]PlaceDetailsResp, 0),
+		PlaceDetails:      make([]PlaceDetailsResp, len(cachePlanSolution.PlaceIDs)),
 		ShownActive:       make([]bool, 0),
 		TravelDestination: destination,
 		TravelDate:        travelDate,
 		Score:             cachePlanSolution.Score,
 		ScoreOld:          cachePlanSolution.ScoreOld,
 	}
+
+	wg := sync.WaitGroup{}
```
Comment (timwangmusic) [inline planner/planner.go:462]: Nit: wg.Add(Len(cachedSolutions))

**Review by timwangmusic — APPROVED** [review]
great work!

File: planner/planner.go (hunk 2 above)
Comment (rwangsc18) [reply planner/planner.go:462]: Done.

**Pull request merged**